build.sh

```diff
@@ -75,10 +75,12 @@ DOCKER_OPTS="${DOCKER_OPTS-$CACHE}"
 
 echo "🐳 Building the Docker image '${DOCKER_TAG}' from the url '${URL}'."
 docker build -t "${DOCKER_TAG}" --build-arg "BRANCH=${BRANCH}" --build-arg "URL=${URL}" --pull ${DOCKER_OPTS} .
-echo "✅ Finished building the Docker images '${DOCKER_TAG}'"
+docker build -t "${DOCKER_TAG}-ldap" --build-arg "BRANCH=${BRANCH}" --build-arg "URL=${URL}" --pull ${DOCKER_OPTS} -f ldap.Dockerfile .
+echo "✅ Finished building the Docker images '${DOCKER_TAG}' and '${DOCKER_TAG}-ldap'"
 
 if [ "${2}" == "--push" ] ; then
-  echo "⏫ Pushing '${DOCKER_TAG}"
+  echo "⏫ Pushing '${DOCKER_TAG} and '${DOCKER_TAG}-ldap'"
   docker push "${DOCKER_TAG}"
-  echo "✅ Finished pushing the Docker image '${DOCKER_TAG}'."
+  docker push "${DOCKER_TAG}-ldap"
+  echo "✅ Finished pushing the Docker images '${DOCKER_TAG}' and '${DOCKER_TAG}-ldap'."
```

Comment:
Instead of duplicating the logic, it would be nicer if you could introduce a variable (e.g. `DOCKERFILE`) that configures which Dockerfile is built, and which would by default be set to 'Dockerfile', but could be overwritten when calling the script.

In order to not mess around with the caching logic, I'd further suggest a `DOCKER_TAG_SUFFIX` variable, which is empty by default, but would be set to `-ldap` for the ldap build.

Reply:
In order to select the Docker image version corresponding to the one being built, it would also be required to pass `TAG` as [`--build-arg`][build-arg] to the `docker build` command.

[build-arg]: https://docs.docker.com/engine/reference/builder/#arg